testbeam_analysis/result_analysis.py

```diff
@@ -1299,3 +1299,273 @@ def histogram_track_angle(input_tracks_file, input_alignment_file=None, output_t
+                intersection_y_cs_1 = intersection_y[np.where(tracks_chunk[:]['n_hits_dut_%i' % actual_dut] == 1)]
+                intersection_z_cs_1 = intersection_z[np.where(tracks_chunk[:]['n_hits_dut_%i' % actual_dut] == 1)]
+                intersection_x_cs_2 = intersection_x[np.where(tracks_chunk[:]['n_hits_dut_%i' % actual_dut] == 2)]
+                intersection_y_cs_2 = intersection_y[np.where(tracks_chunk[:]['n_hits_dut_%i' % actual_dut] == 2)]
+                intersection_z_cs_2 = intersection_z[np.where(tracks_chunk[:]['n_hits_dut_%i' % actual_dut] == 2)]
+                intersection_x_cs_3 = intersection_x[np.where(tracks_chunk[:]['n_hits_dut_%i' % actual_dut] == 3)]
+                intersection_y_cs_3 = intersection_y[np.where(tracks_chunk[:]['n_hits_dut_%i' % actual_dut] == 3)]
+                intersection_z_cs_3 = intersection_z[np.where(tracks_chunk[:]['n_hits_dut_%i' % actual_dut] == 3)]
+                intersection_x_cs_4 = intersection_x[np.where(tracks_chunk[:]['n_hits_dut_%i' % actual_dut] == 4)]
+                intersection_y_cs_4 = intersection_y[np.where(tracks_chunk[:]['n_hits_dut_%i' % actual_dut] == 4)]
+                intersection_z_cs_4 = intersection_z[np.where(tracks_chunk[:]['n_hits_dut_%i' % actual_dut] == 4)]
+
+                # transoform to local coordinate system
+                if use_prealignment:
+                    intersection_x_cs_1_local, intersection_y_cs_1_local, intersection_z_cs_1_local = geometry_utils.apply_alignment(intersection_x_cs_1, intersection_y_cs_1, intersection_z_cs_1,
+                                                                                                                                     dut_index=actual_dut,
+                                                                                                                                     prealignment=prealignment,
+                                                                                                                                     inverse=True)
+
+                    intersection_x_cs_2_local, intersection_y_cs_2_local, intersection_z_cs_2_local = geometry_utils.apply_alignment(intersection_x_cs_2, intersection_y_cs_2, intersection_z_cs_2,
+                                                                                                                                     dut_index=actual_dut,
+                                                                                                                                     prealignment=prealignment,
+                                                                                                                                     inverse=True)
+
+                    intersection_x_cs_3_local, intersection_y_cs_3_local, intersection_z_cs_3_local = geometry_utils.apply_alignment(intersection_x_cs_3, intersection_y_cs_3, intersection_z_cs_3,
```

Comment:
Too much copy paste is a sign that it is time for a for loop and/or helper function